source/Variant.cpp

```diff
@@ -0,0 +1,356 @@
+/* Variant.cpp
+Copyright (c) 2019 by Jonathan Steck
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Variant.h"
+
+#include "DataNode.h"
+#include "GameData.h"
+#include "Random.h"
+#include "Ship.h"
+
+#include <algorithm>
+#include <cmath>
+#include <iterator>
+
+using namespace std;
+
+
+
+// Construct and Load() at the same time.
+Variant::Variant(const DataNode &node)
+{
+	Load(node);
+}
+
+
+
+void Variant::Load(const DataNode &node, const bool global)
+{
+	// If this variant is global (i.e. a root node variant stored in GameData),
+	// it must be named.
+	if(global && node.Size() < 2)
+	{
+		node.PrintTrace("No name specified for variant:");
+		return;
+	}
+	if(node.Size() >= 2 && !node.IsNumber(1))
+		name = node.Token(1);
+	else if(global && node.IsNumber(1))
```

Comment:
If this check (IsNumber) and the check on line 40 are moved to before the variant-load in GameData, then there would no longer be a need to give the global-variable as parameter and it would be possible to avoid the load completely if the name is missing.